zeebe/snapshot/src/main/java/io/camunda/zeebe/snapshots/impl/FileBasedReceivedSnapshot.java

```diff
@@ -124,15 +113,24 @@ private void applyInternal(final SnapshotChunk snapshotChunk) throws SnapshotWri
 
     if (snapshotChunk.getChunkName().equals(FileBasedSnapshotStore.METADATA_FILE_NAME)) {
       try {
-        collectMetadata(snapshotChunk.getContent());
+        collectMetadata(snapshotChunk);
       } catch (final IOException e) {
         throw new SnapshotWriteException("Cannot decode snapshot metadata");
       }
     }
   }
 
-  private void collectMetadata(final byte[] content) throws IOException {
-    metadata = FileBasedSnapshotMetadata.decode(content);
+  private void collectMetadata(final SnapshotChunk chunk) throws IOException {
```

Comment:
The test you added just verifies that we can receive chunked files but not metadata files specifically. Since the implementation is a bit different, I'd like to see tests covering metadata files specifically.

Reply:
The test does send over the `zeebe.metadata` I think it is implied it works do you just want to see an explicit test? 
![Uploading image.png…]()

Reply:
I see. An explicit test would have been nice but I'm okay with this as is.